src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -98,8 +101,25 @@ public PreparedQuery getBigBusinessFromDatabase(){
     }  
     return allBusinesses;
   }
+  
+  private String getUrlFromPlaceDetails(GeoApiContext context, String placeId) {
+    try {
+      PlaceDetails result = new PlaceDetailsRequest(context)
+            .placeId(placeId)
+            .await();
+      if (result.website != null) {
+        return (result.website.toString());
+      }
+      else {
+        return (result.url.toString());
+      }
+    } catch(Exception e) {
+      LOGGER.warning(e.getMessage());
+    }
+    return "NO URL";
```

Comment:
Hmm. What is the expected flow for this failure case? This is still returning a string, just as if everything is ok, and the call to obtain a url succeeded. How is this being handled?

Reply:
The only way that this return value would be reached is if the try fails, so the warning message would be sent from the error. I changed the return value to an empty string and on the front end, if the website is empty, there will be a website unavailable button message. Is that better for the failure case?